mne_qt_browser/_pg_figure.py

```diff
@@ -2206,24 +2225,60 @@ def _region_changed(self):
         )
         onset = np.min(regions_[:, 0])
         offset = np.max(regions_[:, 1])
+        logger.debug(f"New {self.description} region: {onset:.2f} - {offset:.2f}")
         # remove overlapping regions
         for region in overlapping_regions:
             self.weakmain()._remove_region(region, from_annot=False)
         # re-set while blocking the signal to avoid re-running this function
         with SignalBlocker(self):
             self.setRegion((onset, offset))
         self.update_label_pos()
+        # Update the FillBetweenItem shapes for channel specific annotations
+        if self.ch_annot_fills:
+            self._update_channel_annot_fills(onset, offset)
```

Comment:
Let the iterator just return on its own (be a no-op) rather than checking its length
```suggestion
        self._update_channel_annot_fills(onset, offset)
```